From 6c4b5e1765e05ed518854ad6ecc6d48abd219849 Mon Sep 17 00:00:00 2001
From: Lichtenshtein <altruumann@protonmail.com>
Date: Thu, 12 Dec 2025 17:26:09 +0300
Subject: [PATCH] Packages like gettext need to be recompiled or builder won't find translations

Also ffmpeg unable to support opus after applying ffmpeg patch.

---
 .github/workflows/private-windows-x64-build.yml |  3 +--
 dist/win/bundle.sh                              | 10 +++++-----
 dist/win/mxe.diff                               | 14 +++++++-------
 3 files changed, 13 insertions(+), 14 deletions(-)

diff --git a/.github/workflows/caching.yml b/.github/workflows/caching.yml
new file mode 100644
index 0000000..c7b0ec8
--- /dev/null
+++ b/.github/workflows/caching.yml
@@ -0,0 +1,29 @@
+name: Caching
+on:
+  workflow_dispatch:
+  # schedule:
+    # - cron: "0 6 * * 3,6"
+jobs:
+  Windows-MSYS2-MINGW64-Caching:
+    name: Windows MSYS2/MINGW64 Caching
+    runs-on: windows-latest
+    defaults:
+      run:
+        shell: msys2 {0}
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v4
+      - name: Setup MSYS2
+        uses: msys2/setup-msys2@v2
+        with:
+          msystem: mingw64
+          location: C:\msys2
+      - name: Install dependencies
+        run: |
+          cd '${{ github.workspace }}' && ./dist/win/install_deps.sh
+          du -sh 'C:\msys2'
+      - name: Cache MSYS2
+        uses: actions/cache@v4
+        with:
+          key: spek-x-msys2-mingw64
+          path: C:\msys2
diff --git a/.github/workflows/private-windows-x64-build.yml b/.github/workflows/private-windows-x64-build.yml
index cab0ee0..08a9174 100644
--- a/.github/workflows/private-windows-x64-build.yml
+++ b/.github/workflows/private-windows-x64-build.yml
@@ -43,8 +43,7 @@ jobs:
         sudo add-apt-repository "deb [arch=amd64] https://pkg.mxe.cc/repos/apt focal main"
         sudo apt-get update
         sudo apt-get -y install mxe-x86-64-w64-mingw32.static-cc
-        sudo apt-get -y install autoconf automake autopoint bash bison bzip2 flex g++ g++-multilib gettext git gperf intltool libc6-dev-i386 libgdk-pixbuf2.0-dev libltdl-dev libgl-dev libssl-dev libtool-bin libxml-parser-perl lzip make openssl p7zip-full patch perl python3 python3-mako python3-pkg-resources ruby sed unzip wget xz-utils
-        sudo apt-get -y install binutils brotli bzip2 ccache cmake dbus expat fontconfig gcc gettext lame libxml2 meson mpc pkgconf speex yasm zstd
+        sudo apt-get -y install autoconf automake autopoint bash bison bzip2 flex g++ g++-multilib gettext git gperf intltool libc6-dev-i386 libgdk-pixbuf2.0-dev libltdl-dev libgl-dev libssl-dev libtool-bin libxml-parser-perl lzip make openssl p7zip-full patch perl python3 python3-mako python3-pkg-resources ruby sed unzip wget xz-utils mingw-w64
 
     - name: Clone, patch, and build MXE - Spek Dependencies
       if: ${{ steps.cache-bin.outputs.cache-hit != 'true' }}
@@ -53,7 +52,7 @@ jobs:
         git clone https://github.com/mxe/mxe.git
         cd mxe
         patch -p1 < ../dist/win/mxe.diff || echo "Patch failed, building without modifications"
-        make wxwidgets pthreads ffmpeg EXCLUDE_PKGS="binutils brotli bzip2 ccache cmake dbus expat fontconfig gcc gettext lame libxml2 meson mpc pkgconf speex yasm zstd" -j$(nproc) JOBS=$(nproc) MXE_TARGETS='x86_64-w64-mingw32.static'
+        make wxwidgets pthreads ffmpeg -j$(nproc) JOBS=$(nproc) MXE_TARGETS='x86_64-w64-mingw32.static'
 
     - name: Build Spek
       shell: bash
@@ -63,8 +63,8 @@ jobs:
 
         ./dist/win/bundle.sh
 
-        find . -iname "config.log"
-        cat ./config.log
+        # find . -iname "config.log"
+        # cat ./config.log
 
     - name: Upload artifact
       uses: actions/upload-artifact@v4
diff --git i/dist/win/bundle.sh w/dist/win/bundle.sh
index f3465f7..b8244f3 100755
--- i/dist/win/bundle.sh
+++ w/dist/win/bundle.sh
@@ -12,9 +12,10 @@ MAKE=make
 ZIP=zip
 
 HOST=x86_64-w64-mingw32.static
-LANGUAGES="bs ca cs da de el eo es fi fr gl he hr hu id it ja ko lv nb nl nn pl pt_BR ru sk sr@latin sv th tr uk vi zh_CN zh_TW"
+LANGUAGES="ru"
 PATH="$MXE"/bin:$PATH
 WX_CONFIG="$MXE"/"$HOST"/bin/wx-config
+STRIP="$HOST"-strip
 
 cd $(dirname $0)/../..
 rm -fr dist/win/build && mkdir dist/win/build
@@ -33,6 +34,8 @@ LDFLAGS="-mwindows dist/win/spek.res" ./autogen.sh \
     "$MAKE" -j$(nproc) JOBS=$(nproc) && \
     "$MAKE" install || exit 1
 
+"$STRIP" dist/win/build/bin/spek.exe
+
 # Copy files to the bundle
 cd dist/win
 rm -fr Spek && mkdir Spek
diff --git i/dist/win/mxe.diff w/dist/win/mxe.diff
index 2fe2efc..1352470 100644
--- i/dist/win/mxe.diff
+++ w/dist/win/mxe.diff
@@ -2,70 +2,79 @@ diff --git a/src/ffmpeg.mk b/src/ffmpeg.mk
 index b486d859..d1d7013d 100644
 --- a/src/ffmpeg.mk
 +++ b/src/ffmpeg.mk
-@@ -8,9 +8,9 @@ $(PKG)_CHECKSUM := 733984395e0dbbe5c046abda2dc49a5544e7e0e1e2366bba849222ae9e3a0
+@@ -8,9 +8,7 @@ $(PKG)_CHECKSUM := 733984395e0dbbe5c046abda2dc49a5544e7e0e1e2366bba849222ae9e3a0
  $(PKG)_SUBDIR   := $(PKG)-$($(PKG)_VERSION)
  $(PKG)_FILE     := $(PKG)-$($(PKG)_VERSION).tar.xz
  $(PKG)_URL      := https://ffmpeg.org/releases/$($(PKG)_FILE)
 -$(PKG)_DEPS     := cc bzip2 gnutls lame libass libbluray libbs2b libcaca \
 -                   libvpx opencore-amr opus sdl2 speex theora vidstab \
 -                   vo-amrwbenc vorbis x264 x265 xvidcore yasm zlib
-+$(PKG)_DEPS     := cc bzip2 \
-+                   sdl2 \
-+                   yasm zlib
++$(PKG)_DEPS     := cc bzip2 lame libass opus sdl2 vorbis yasm zlib
  
  # DO NOT ADD fdk-aac OR openssl SUPPORT.
  # Although they are free softwares, their licenses are not compatible with
-@@ -38,29 +38,34 @@ define $(PKG)_BUILD
+@@ -32,36 +30,43 @@ define $(PKG)_BUILD
+         --arch=$(firstword $(subst -, ,$(TARGET))) \
+         --target-os=mingw32 \
+         --prefix='$(PREFIX)/$(TARGET)' \
+-        $(if $(BUILD_STATIC), \
+-            --enable-static --disable-shared , \
+-            --disable-static --enable-shared ) \
          --yasmexe='$(TARGET)-yasm' \
++        --disable-avdevice \
++        --disable-avfilter \
++        --disable-bsfs \
          --disable-debug \
-         --disable-pthreads \
+-        --disable-pthreads \
 -        --enable-w32threads \
++        --disable-decoders \
++        --disable-devices \
          --disable-doc \
-+        --disable-network \
-+        --disable-xlib \
-+        --disable-hwaccels \
-+        --disable-bsfs \
-+        --disable-protocols \
-+        --disable-programs \
-+        --disable-avdevice \
-+        --disable-swresample \
-+        --disable-swscale \
-+        --disable-postproc \
-+        --disable-avfilter \
 +        --disable-encoders \
-+        --disable-muxers \
-+        --disable-devices \
 +        --disable-filters \
++        --disable-hwaccels \
++        --disable-muxers \
++        --disable-network \
 +        --disable-parsers \
-+        --disable-decoders \
-+        --enable-decoder=aac,aac_fixed,aac_latm,ac3,ac3_fixed,acelp_kelvin,alac,als,amrnb,amrwb,ape,aptx,aptx_hd,atrac1,atrac3,atrac3al,atrac3p,atrac3pal,atrac9,binkaudio_dct,binkaudio_rdft,bmv_audio,cook,dca,dfpwm,dolby_e,dsd_lsbf,dsd_msbf,dsd_lsbf_planar,dsd_msbf_planar,dsicinaudio,dss_sp,dst,eac3,evrc,fastaudio,ffwavesynth,flac,g723_1,g729,gsm,gsm_ms,hca,hcom,iac,ilbc,imc,interplay_acm,mace3,mace6,metasound,mlp,mp1,mp1float,mp2,mp2float,mp3float,mp3,mp3adufloat,mp3adu,mp3on4float,mp3on4,mpc7,mpc8,msnsiren,nellymoser,on2avc,opus,paf_audio,qcelp,qdm2,qdmc,ra_144,ra_288,ralf,sbc,shorten,sipr,siren,smackaud,sonic,tak,truehd,truespeech,tta,twinvq,vmdaudio,vorbis,wavpack,wmalossless,wmapro,wmav1,wmav2,wmavoice,ws_snd1,xma1,xma2,pcm_alaw,pcm_bluray,pcm_dvd,pcm_f16le,pcm_f24le,pcm_f32be,pcm_f32le,pcm_f64be,pcm_f64le,pcm_lxf,pcm_mulaw,pcm_s8,pcm_s8_planar,pcm_s16be,pcm_s16be_planar,pcm_s16le,pcm_s16le_planar,pcm_s24be,pcm_s24daud,pcm_s24le,pcm_s24le_planar,pcm_s32be,pcm_s32le,pcm_s32le_planar,pcm_s64be,pcm_s64le,pcm_sga,pcm_u8,pcm_u16be,pcm_u16le,pcm_u24be,pcm_u24le,pcm_u32be,pcm_u32le,pcm_vidc,derf_dpcm,gremlin_dpcm,interplay_dpcm,roq_dpcm,sdx2_dpcm,sol_dpcm,xan_dpcm,adpcm_4xm,adpcm_adx,adpcm_afc,adpcm_agm,adpcm_aica,adpcm_argo,adpcm_ct,adpcm_dtk,adpcm_ea,adpcm_ea_maxis_xa,adpcm_ea_r1,adpcm_ea_r2,adpcm_ea_r3,adpcm_ea_xas,adpcm_g722,adpcm_g726,adpcm_g726le,adpcm_ima_acorn,adpcm_ima_amv,adpcm_ima_alp,adpcm_ima_apc,adpcm_ima_apm,adpcm_ima_cunning,adpcm_ima_dat4,adpcm_ima_dk3,adpcm_ima_dk4,adpcm_ima_ea_eacs,adpcm_ima_ea_sead,adpcm_ima_iss,adpcm_ima_moflex,adpcm_ima_mtf,adpcm_ima_oki,adpcm_ima_qt,adpcm_ima_rad,adpcm_ima_ssi,adpcm_ima_smjpeg,adpcm_ima_wav,adpcm_ima_ws,adpcm_ms,adpcm_mtaf,adpcm_psx,adpcm_sbpro_2,adpcm_sbpro_3,adpcm_sbpro_4,adpcm_swf,adpcm_thp,adpcm_thp_le,adpcm_vima,adpcm_xa,adpcm_yamaha,adpcm_zork \
-+        --enable-w32threads \
-+        --enable-pic \
-+        --enable-protocol=file,pipe \
-+        --enable-parser=aac,aac_latm,ac3,cook,dca,dirac,dolby_e,flac,g723_1,g729,gsm,mlp,mpegaudio,opus,sbc,sipr,tak,vorbis \
++        --disable-postproc \
++        --disable-programs \
++        --disable-protocols \
++        --disable-pthreads \
++        --disable-shared \
++        --disable-swscale \
++        --disable-xlib \
++        --enable-decoder=aac,aac_fixed,aac_latm,ac3,ac3_fixed,acelp_kelvin,alac,als,amrnb,amrwb,ape,aptx,aptx_hd,atrac1,atrac3,atrac3al,atrac3p,atrac3pal,atrac9,binkaudio_dct,binkaudio_rdft,bmv_audio,cook,dca,dfpwm,dolby_e,dsd_lsbf,dsd_msbf,dsd_lsbf_planar,dsd_msbf_planar,dsicinaudio,dss_sp,dst,eac3,evrc,fastaudio,ffwavesynth,flac,g723_1,g729,gsm,gsm_ms,hca,hcom,iac,ilbc,imc,interplay_acm,mace3,mace6,metasound,mlp,mp1,mp1float,mp2,mp2float,mp3float,mp3,mp3adufloat,mp3adu,mp3on4float,mp3on4,mpc7,mpc8,msnsiren,nellymoser,on2avc,opus,paf_audio,qcelp,qdm2,qdmc,ra_144,ra_288,ralf,sbc,shorten,sipr,siren,smackaud,sonic,tak,truehd,truespeech,tta,twinvq,vmdaudio,vorbis,wavpack,wmalossless,wmapro,wmav1,wmav2,wmavoice,ws_snd1,xma1,xma2,pcm_alaw,pcm_bluray,pcm_dvd,pcm_f16le,pcm_f24le,pcm_f32be,pcm_f32le,pcm_f64be,pcm_f64le,pcm_lxf,pcm_mulaw,pcm_s8,pcm_s8_planar,pcm_s16be,pcm_s16be_planar,pcm_s16le,pcm_s16le_planar,pcm_s24be,pcm_s24daud,pcm_s24le,pcm_s24le_planar,pcm_s32be,pcm_s32le,pcm_s32le_planar,pcm_s64be,pcm_s64le,pcm_sga,pcm_u8,pcm_u16be,pcm_u16le,pcm_u24be,pcm_u24le,pcm_u32be,pcm_u32le,pcm_vidc,derf_dpcm,gremlin_dpcm,interplay_dpcm,roq_dpcm,sdx2_dpcm,sol_dpcm,xan_dpcm,adpcm_4xm,adpcm_adx,adpcm_afc,adpcm_agm,adpcm_aica,adpcm_argo,adpcm_ct,adpcm_dtk,adpcm_ea,adpcm_ea_maxis_xa,adpcm_ea_r1,adpcm_ea_r2,adpcm_ea_r3,adpcm_ea_xas,adpcm_g722,adpcm_g726,adpcm_g726le,adpcm_ima_acorn,adpcm_ima_amv,adpcm_ima_alp,adpcm_ima_apc,adpcm_ima_apm,adpcm_ima_cunning,adpcm_ima_dat4,adpcm_ima_dk3,adpcm_ima_dk4,adpcm_ima_ea_eacs,adpcm_ima_ea_sead,adpcm_ima_iss,adpcm_ima_moflex,adpcm_ima_mtf,adpcm_ima_oki,adpcm_ima_qt,adpcm_ima_rad,adpcm_ima_ssi,adpcm_ima_smjpeg,adpcm_ima_wav,adpcm_ima_ws,adpcm_ms,adpcm_mtaf,adpcm_psx,adpcm_sbpro_2,adpcm_sbpro_3,adpcm_sbpro_4,adpcm_swf,adpcm_thp,adpcm_thp_le,adpcm_vima,adpcm_xa,adpcm_yamaha,adpcm_zork \
          --enable-gpl \
-         --enable-version3 \
-         --extra-libs='-mconsole' \
+-        --enable-version3 \
+-        --extra-libs='-mconsole' \
 -        --enable-gnutls \
--        --enable-libass \
+         --enable-libass \
 -        --enable-libbluray \
 -        --enable-libbs2b \
 -        --enable-libcaca \
--        --enable-libmp3lame \
+         --enable-libmp3lame \
 -        --enable-libopencore-amrnb \
 -        --enable-libopencore-amrwb \
--        --enable-libopus \
+         --enable-libopus \
 -        --enable-libspeex \
 -        --enable-libtheora \
 -        --enable-libvidstab \
 -        --enable-libvo-amrwbenc \
--        --enable-libvorbis \
+         --enable-libvorbis \
 -        --enable-libvpx \
 -        --enable-libx264 \
 -        --enable-libx265 \
 -        --enable-libxvid \
-         --extra-ldflags="-fstack-protector" \
++        --enable-parser=aac,aac_latm,ac3,cook,dca,dirac,dolby_e,flac,g723_1,g729,gsm,mlp,mpegaudio,opus,sbc,sipr,tak,vorbis \
++        --enable-pic \
++        --enable-protocol=file,pipe \
 +        --enable-small \
-+        --extra-cflags="-march=broadwell -mtune=broadwell -O3"
++        --enable-static \
++        --enable-version3 \
++        --enable-w32threads \
++        --extra-cflags="-march=broadwell -mtune=broadwell -O3" \
+         --extra-ldflags="-fstack-protector" \
++        --extra-libs='-mconsole' \
          $($(PKG)_CONFIGURE_OPTS)
      $(MAKE) -C '$(BUILD_DIR)' -j '$(JOBS)'
+     $(MAKE) -C '$(BUILD_DIR)' -j 1 install
-- 
2.49.0.windows.1

